dotCMS/src/main/java/com/dotcms/contenttype/business/ContentTypeDestroyAPIImpl.java

```diff
@@ -0,0 +1,351 @@
+                        partitionSize)
+        );
+
+        return Lists.partition(new ArrayList<>(contents.entrySet()), partitionSize).stream().map(
+                partition -> partition.stream()
+                        .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue))
+        ).collect(Collectors.toList());
+
+    }
+
+    static class ContentletVersionInfo {
+        final String inode;
+        final Number languageId;
+
+        final String hostInode;
+
+        final String folderInode;
+
+        private ContentletVersionInfo(final String inode, final Number languageId, final String hostInode, final String folderInode) {
+            this.inode = inode;
+            this.languageId = languageId;
+            this.hostInode = hostInode;
+            this.folderInode = folderInode;
+        }
+
+        public static ContentletVersionInfo of(final String inode, final Number languageId, final String hostId, final String folderId) {
+            return new ContentletVersionInfo(inode, languageId, hostId, folderId);
+        }
+
+        @Override
+        public String toString() {
+            return "ContentletVersionInfo{" +
+                    "inode='" + inode + '\'' +
+                    ", languageId=" + languageId +
+                    ", hostInode='" + hostInode + '\'' +
+                    ", folderInode='" + folderInode + '\'' +
+                    '}';
+        }
+
+        public String getInode() {
+            return inode;
+        }
+
+        public Number getLanguageId() {
+            return languageId;
+        }
+
+        public String getHostInode() {
+            return hostInode;
+        }
+        public String getFolderInode() {
+            return null == folderInode ? Host.SYSTEM_HOST : folderInode;
+        }
+
+    }
+
+    /**
+     * Send out a system-wide notification that a content type has been destroyed
+     * @param contentType
+     */
+    private void notifyContentTypeDestroyed(final ContentType contentType) {
+        try {
+            final NotificationAPI notificationAPI = APILocator.getNotificationAPI();
+            // Search for the CMS Admin role and System User
+            final Role cmsAdminRole = APILocator.getRoleAPI().loadCMSAdminRole();
+            final User systemUser = APILocator.systemUser();
+
+            notificationAPI.generateNotification(
+                    new I18NMessage("contenttype.destroy.complete.title"),
+                    new I18NMessage("contenttype.destroy.complete.message", null,
+                            contentType.name()), null,
+                    // no actions
+                    NotificationLevel.WARNING, NotificationType.GENERIC,
+                    Visibility.ROLE, cmsAdminRole.getId(), systemUser.getUserId(),
+                    systemUser.getLocale()
+            );
+        } catch (DotDataException e) {
+            Logger.error(getClass(), e.getMessage(), e);
```

Comment:
you can add the content type variable name here to give more info about the error

Reply:
done